src/OrchardCore/OrchardCore.Users.Core/Services/RoleService.cs

```diff
@@ -44,4 +48,10 @@ public Task<IEnumerable<string>> GetNormalizedRoleNamesAsync()
     {
         return Task.FromResult<IEnumerable<string>>(_roleManager.Roles.Select(a => _roleManager.NormalizeKey(a.RoleName)));
     }
+
+    public ValueTask<bool> IsAdminRoleAsync(string role)
```

Comment:
I'd rather see `SystemRolesCatalog` internal and make public a method to return the admin role name, and maybe the system role names.

Reply:
That can be in addition to this, not to have to repeat the role name matching all over the place (which was the case before, and can add uncertainty to whether it's done in a case-sensitive way or not, and with which culture).

Reply:
`SystemRolesCatalog` was renamed with `DefaultSystemRoleNameProvider` which implements a new interface `ISystemRoleNameProvider`.